eth/stagedsync/stage_mining_create_block.go

```diff
@@ -174,17 +174,9 @@ func SpawnMiningCreateBlockStage(s *StageState, tx kv.RwTx, cfg MiningCreateBloc
 	}
 
 	header := core.MakeEmptyHeader(parent, &cfg.chainConfig, timestamp, &cfg.miner.MiningConfig.GasLimit)
-	if cfg.blockBuilderParameters != nil && cfg.blockBuilderParameters.GasLimit != nil {
-		log.Info("MMDBG Override gas limit from Engine API", "old", header.GasLimit, "new", *cfg.blockBuilderParameters.GasLimit)
+	if cfg.blockBuilderParameters != nil && cfg.blockBuilderParameters.GasLimit != nil && header.GasLimit != *cfg.blockBuilderParameters.GasLimit {
```

Comment:
The previous check ensured that the GasLimit was not nil before dereferencing -- are we risking a nil dereference there?